src/static/bitbucket/components/App.tsx

```diff
@@ -56,6 +55,12 @@ const App = () => {
   const [queue, setQueue] = useState<QueueResponse['queue'] | undefined>();
   const [_, setLoadStatus, loadStatusRef] = useState<LoadStatus>('not-loaded');
   const [state, dispatch] = useState(initialState);
+  const widgetSettings = useWidgetSettings();
+  const widgetSettingsRef = useRef(widgetSettings);
+
+  if (widgetSettings !== widgetSettingsRef.current) {
```

Comment:
dumb question: why is this needed?

Reply:
This was slightly confusing to me as well (been out of touch with react for an year), but refs are needed because I'm reading these values inside a setTimeout call — which saves its own copy of variables in its closure. Using a ref (and keeping it up to date) ensures that only the latest value is used, even if we're inside a closure.